src/mixins/keyboard-shortcut-mixin.ts

```diff
@@ -1,26 +1,65 @@
 import { LitElement } from "lit-element";
 import { Constructor } from "../types";
 
+type RegisteredShortcuts = {
+  [key in HAKeyboardShortcut]?: () => void;
+};
+
+export enum HAKeyboardShortcut {
+  CTRL_S = "CtrlS",
+  CTRL_P = "CtrlP",
+  CTRL_SHIFT_P = "CtrlShiftP",
+}
+
+const isMac = /(Mac|iPhone|iPod|iPad)/i.test(navigator.platform);
+
 export const KeyboardShortcutMixin = <T extends Constructor<LitElement>>(
   superClass: T
 ) =>
   class extends superClass {
-    private _keydownEvent = (event: KeyboardEvent) => {
-      if ((event.ctrlKey || event.metaKey) && event.key === "s") {
-        event.preventDefault();
-        this.handleKeyboardSave();
-      }
-    };
+    private _registeredShortcuts: RegisteredShortcuts = {};
 
-    public connectedCallback() {
-      super.connectedCallback();
-      this.addEventListener("keydown", this._keydownEvent);
+    protected registerShortcut(
+      shortcut: HAKeyboardShortcut,
+      callback: () => void,
+      element: HTMLElement | Document = this
+    ) {
+      element.addEventListener("keydown", this._keydownEvent);
+      this._registeredShortcuts[shortcut] = callback;
     }
 
-    public disconnectedCallback() {
-      this.removeEventListener("keydown", this._keydownEvent);
```

Comment:
Why did we remove this?

Reply:
Not sure. I probably just saw that `connectedCallback` wasn't needed anymore so I reflexively remove disconnected as well.

I put it back.